enum_methods.rb

```diff
@@ -0,0 +1,155 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    each do |i|
+      yield(i)
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    my_each do |element|
+      yield(element, i)
+      i += 1
+    end
```

Comment:
Once you change the above, is possible that this won't work anymore, so you will need to check it. If there is a problem, just implement another loop here.